### PR TITLE
MYRIAD-162

### PR DESCRIPTION
**Pull request by DarinJ**
This corrects how offers are handled when using the framework role parameter.  Prior to this fix if Myriad received an offer with resource from the default role and it's declared role it incorrectly declared all cpu and memory as used by the default role.  This let to TASK_LOST errors, which the logs stated were a result of insufficient resources.

Tested on a cluster with a variety of different agent configurations, which required the use of cpus and mem resources from both the default ("*") and frameworkRole ("roleA").  Verified correct behavior for both coarse and fine grained scaling.

NB: This PR does not fix the issue for the ports resource and explicitly forces Myriad the use the default ("*") role for ports.  I believe there is potential to use reserved port for a role to help Myriad, but that would require more changes and be more of a improvement than bug-fix, hence is to be future work.

I would like to squash commits and clean up formatting before merge, but I thought I'd submit the PR now to solicit comments.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Wed Jun 10 10:58:12 CDT 2015
+#Wed Oct 28 13:11:41 EDT 2015
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.4-all.zip
```
Comment (smarella) [inline gradle/wrapper/gradle-wrapper.properties]: I think this file shouldn't be part of the PR.

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceTaskFactoryImpl.java [hunk 2]
```diff
@@ -126,13 +112,13 @@ public TaskInfo createTask(Offer offer, FrameworkID frameworkId, TaskID taskId,
...
     LOGGER.info("Command line for service: {} is: {}", nodeTask.getTaskPrefix(), strB.toString());
 
-    Scalar taskMemory = Scalar.newBuilder().setValue(nodeTask.getProfile().getMemory()).build();
-    Scalar taskCpus = Scalar.newBuilder().setValue(nodeTask.getProfile().getCpus()).build();
-
     TaskInfo.Builder taskBuilder = TaskInfo.newBuilder();
 
-    taskBuilder.setName(nodeTask.getTaskPrefix()).setTaskId(taskId).setSlaveId(offer.getSlaveId()).addResources(Resource.newBuilder().setName("cpus").setType(Value.Type.SCALAR).setScalar(taskCpus).build()).addResources(Resource.newBuilder()
-        .setName("mem").setType(Value.Type.SCALAR).setScalar(taskMemory).build());
+    taskBuilder.setName(nodeTask.getTaskPrefix())
```
Comment (smarella) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceTaskFactoryImpl.java]: We can avoid a complex merge if leave out the formatting to @klucar. He is already doing it in another PR.

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java [hunk 3]
```diff
@@ -189,23 +175,51 @@ public TaskInfo createTask(Offer offer, FrameworkID frameworkId, TaskID taskId,
...
+                  .addRange(Range.newBuilder()
+                      .setBegin(ports.getWebAppHttpPort())
+                      .setEnd(ports.getWebAppHttpPort())
+                      .build())
+                  .addRange(Range.newBuilder()
+                      .setBegin(ports.getShufflePort())
+                      .setEnd(ports.getShufflePort())
+                      .build())))
+          .setExecutor(executorInfo)
+          .build();
```
Comment (smarella) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java]: Same as above. If this is just a formatting change, I suggest we leave it to @klucar to avoid conflicts while merging the PRs.

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java [hunk 4]
```diff
@@ -110,28 +96,28 @@ public NMTaskFactoryImpl(MyriadConfiguration cfg, TaskUtils taskUtils, ExecutorC
...
     }
-    
+
     //Utility function to get the first NMPorts.expectedNumPorts number of ports of an offer
     @VisibleForTesting
     protected static NMPorts getPorts(Offer offer) {
       HashSet<Long> ports = new HashSet<>();
       for (Resource resource : offer.getResourcesList()) {
-        if (resource.getName().equals("ports")) {
+        if (resource.getName().equals("ports") && (!resource.hasRole() || resource.getRole().equals("*"))) {
```
Comment (smarella) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java]: shouldn't we check the framework role for ports as well?

Comment (smarella) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java]: Oh.. now I see that you mentioned the ports are obtained from * role and not from the framework role. Can you please open a JIRA to track this for future?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (DarinJ) [inline gradle/wrapper/gradle-wrapper.properties]: yeah happens everytime, I usually fix it prior. Will fix.

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceTaskFactoryImpl.java (hunk 2 above)
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceTaskFactoryImpl.java]: Lines 120 and 121 were the only lines here that aren't formatting, I could remove the formatting.

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java (hunk 4 above)
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java]: Was planning on it.  I felt we should treat the framework role ports as special as we could reserve them for thing that need specific ports (JHS, etc).

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java (hunk 3 above)
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java]: 185 and 186 are not formatting.  The rest is.

**Comment by DarinJ** [comment]
@smarella looks like all comments are formatting related.  How about I just rebase since you've merged @klucar's changes?  And I need to fix my imports.

**Comment by smarella** [comment]
Sure. Please rebase.

Santosh

**Comment by smarella** [comment]
Ship It! :rocket: 
Sanity tested on a dev cluster. Didn't see any regressions with the "default" role (*).

**Comment by smarella** [comment]
Merged: https://github.com/apache/incubator-myriad/commit/2e14b4ba362607e6fd4cd22adf0973a83ab25416
